iterative_sorting/iterative_sorting.py

```diff
@@ -1,20 +1,20 @@
 # TO-DO: Complete the selection_sort() function below
 def selection_sort(arr):
-    # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
-        smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc)
-
-        # TO-DO: swap
-
+        smallest = min(arr[i:])
+        smallest_index = arr.index(smallest)
+        if smallest < arr[cur_index]:
+            arr[cur_index], arr[smallest_index] = arr[smallest_index], arr[cur_index]
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort(arr):
-
+    for i in range(0, len(arr) - 1):
```

Comment:
This sorts properly, but also keep in mind there's another implementation of bubble sort! The other implementation pattern involves using a while loop and setting flags inside and outside of the loop to determine if sorting has been completed before loop back through the array. This pattern is used very often to solve problems that require iteration and will come up often in future projects. Let me know if you ever want to discuss that pattern more deeply.